tests/s-ucontext.c

```diff
@@ -36,7 +31,6 @@ int main(int argc, char *argv[])
 	makecontext(&new, (void (*)(void))bar, 1, n);
 
 	foo(&curr, &new);
-	n = baz(n);
```

Comment:
This was tricky to keep `baz()` call here in clang with optimization because `baz()` call is optimized away by clang constant folding, which can't be disabled with `-fno-inline`.

I've just made `baz()` logic a little bit different that prevents the optimization. This is not guaranteed but works. Otherwise, we should take two different approaches.

1. adding `asm("")` inside `baz()` function
2. adding the following clang only attribute to `baz()` function.
```c
#if __clang__
__attribute__((optnone))
#endif
```
The above both approaches also fix the problem, but I rather slightly changed the logic in `baz()`. Please let me know if you prefer approach 1 or 2. Otherwise, let's go with the current approach.

Reply:
If we add it in fixup, then how do we catch when it doesn’t call `baz` properly?

> This was to check if it shows any function after the context switch properly.

Reply:
Well.. only if it's not optimized out. :)  Yep, surely it'd be nice to keep it.

Reply:
> How about making the argument c as volatile?

Thanks. It also fixes the problem so I make `baz()` back to the original, but just added a `volatile` keyword to `c`.